src/ucs/datastruct/queue.h

```diff
@@ -155,12 +166,14 @@ static inline ucs_queue_elem_t *ucs_queue_pull(ucs_queue_head_t *queue)
  * @param new_elems Queue of elements to add.
  */
 static inline void ucs_queue_splice(ucs_queue_head_t *queue,
-                                    ucs_queue_head_t *new_elems)
+                                    const ucs_queue_head_t *new_elems)
 {
+    ucs_queue_head_t temp = *new_elems;
+
     if (!ucs_queue_is_empty(new_elems)) {
-        *queue->ptail = new_elems->head;
-        queue->ptail = new_elems->ptail;
-        new_elems->ptail = &new_elems->head;
+        *queue->ptail = temp.head;
+        queue->ptail = temp.ptail;
+        temp.ptail = &temp.head;
```

Comment:
we should not change splice function, but change its usage to introduce local var